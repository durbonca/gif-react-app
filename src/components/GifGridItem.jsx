import React from 'react'
import PropTypes from 'prop-types'

export const GifGridItem = ( { title, url } ) => {
  return (
    <div className='card animate__animated animate__fadeIn'>
      <img src={url} alt={title}/>
      <div className='title-box'>
        <p>{title}</p>
      </div>
    </div>
  )
}

GifGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
}