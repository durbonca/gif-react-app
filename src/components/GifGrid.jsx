import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs'
import { GifGridItem } from './GifGridItem'
import PropTypes from 'prop-types'

export const GifGrid = ({category}) => {

    const { data:images, loading } = useFetchGifs( category );

    return (
        <>
            <h3 className='animate__animated animate__fadeIn'>{category}</h3>

            { loading && <i className='animate__animated flash'>Cargando...</i>}

            <div className='card-grid'>
                {images.map( ( img, i ) =>
                    (
                        <GifGridItem key={i} {...img} />
                        )
                        )}
            </div>
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}