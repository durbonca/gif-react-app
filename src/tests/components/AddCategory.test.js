import { shallow } from "enzyme";
import { AddCategory } from "../../components/AddCategory";

describe('<AddCategory /> Test', () => {

    const handleAdd = jest.fn();
    let wrapper = shallow(<AddCategory handleAdd={handleAdd} />);

    beforeEach(() => {
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory handleAdd={handleAdd} />);
    })

    test('Verificar Snapshot',()=>{
        expect(wrapper).toMatchSnapshot();
    })

    test('debe actualizar caja texto', ()=>{
        const input = wrapper.find('input');
        const value = 'Hola Mundo';
        input.simulate('change', {target: { value }} )
        expect(wrapper.find('p').text().trim()).toBe(value);
    })

    test('No debe de postear la informacion onSubmit', () => {
        wrapper.find('form').simulate('submit', { preventDefault(){} });
        expect(handleAdd).not.toHaveBeenCalled();
    })

    test('debe de llamar el setCategories y limpiar la caja de texto', () => {
        const value = 'prueba';
        wrapper.find('input').simulate('change', { target: { value } })
        wrapper.find('form').simulate('submit', { preventDefault(){} })
        expect(handleAdd).toHaveBeenCalled();
        expect(handleAdd).toHaveBeenCalledWith(expect.any(Object));
        expect(wrapper.find('input').prop('value')).toBe('')
    })
})