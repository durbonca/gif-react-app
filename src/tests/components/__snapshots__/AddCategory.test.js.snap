// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddCategory /> Test Verificar Snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
  <button
    type="submit"
  >
    Agregar
  </button>
</form>
`;
